### PR TITLE
Added function to validated if the

### PR DESCRIPTION
**Pull request by LiliamLeme**
Added function to validated if the 
password has hyphen.
Function will validate ADLS and ADF password 
for SP creation  during the infra deployment

# Type of PR
Code changes

- Documentation changes
- Code changes
- Test changes
- CI-CD changes
- GitHub Template changes

## Purpose

- ...
If the password created has an hyphen it will break the code parameterization. the purposed of the fix is validated the password created and retry in case has an hyphen, which is random issue.

## Author pre-publish checklist

- [ X] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

- ...

## Issues Closed or Referenced

- Closes #issue_number

- References #issue_number

### CODE REVIEW

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 1]
```diff
@@ -175,27 +175,73 @@ appinsights_connstr=$(az monitor app-insights component show \
...
+    local retry_count="$3"
+
+    if [[ -z "$password" || "$password" == "-" ]]; then
+        ###if there is an hyphen retry
+        ((retry_count++))
+        if [[ $retry_count -ge $max_retries ]]; then
+            echo "Error: Failed to generate a valid password after $max_retries retries." >&2
+            exit 1
+        fi
+        echo "Invalid password as the hyphen is the first character. Retrying... Attempt $retry_count"
```
Comment (thesqlpro) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:193]: Minor rewording
Original
"Invalid password as the hyphen is the first character." ....
Suggestion
"Invalid password. The first character cannot be a hyphen." ....

**Review by thesqlpro — APPROVED** [review]
Looks good. Just suggested a minor rewording for an error message that is displayed to the user running the code.

**Review by ydaponte — CHANGES_REQUESTED** [review]
Please change the PR to merge to the feature branch first instead of directly to main: [feat/e2e-databricks-updates](https://github.com/Azure-Samples/modern-data-warehouse-dataops/tree/feat/e2e-databricks-updates)

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh (hunk 1 above)
Comment (LiliamLeme) [reply e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:193]: it would be better in fact

**Comment by LiliamLeme** [comment]
> Please change the PR to merge to the feature branch first instead of directly to main: [feat/e2e-databricks-updates](https://github.com/Azure-Samples/modern-data-warehouse-dataops/tree/feat/e2e-databricks-updates)

done @ydaponte

**Pull request merged**